symengine/subs.h

```diff
@@ -99,10 +99,25 @@ class SubsVisitor : public BaseVisitor<SubsVisitor>
     {
         RCP<const Basic> base_new = apply(x.get_base());
         RCP<const Basic> exp_new = apply(x.get_exp());
-        if (base_new == x.get_base() and exp_new == x.get_exp())
+        if (subs_dict_.size() == 1 and is_a<Pow>(*((*subs_dict_.begin()).first))
+            and not is_a<Add>(
+                    *down_cast<const Pow &>(*(*subs_dict_.begin()).first)
+                         .get_exp())) {
```

Comment:
At the moment, `subs` in symengine is like `xreplace` in sympy and doesn't do these kind of simplifications. If you want to do this, I would rename the existing method to `xreplace` (`XReplaceVisitor`) and then use `subs` (`SubsVisitor` inheriting from `XReplaceVisitor`) for the new implementation.